internal/feature_test.go

```diff
@@ -193,69 +194,224 @@ func testDisconnectWhileRunningOperation(feature *ScriptBasedSoftwareUpdatable,
+	w.Host(true, false, "", "")
+	w.AddInstallScript()
+	defer w.Close()
+
+	installScript := w.GenerateSoftwareArtifacts(false, "install")
+	artifacts := []*hawkbit.SoftwareArtifactAction{
+		convertLocalArtifact(aPath, a, aHash, len(aBody)),
+		convertLocalArtifact(bPath, b, bHash, len(bBody)),
+		installScript[0],
+	}
+
+	testDownloadInstall(feature, mc, artifacts, true, b, t)
+}
+
+// TestScriptBasedDownloadAndInstallLocalResources tests ScriptBasedSoftwareUpdatable core functionality: init, install and download operations,
+// but working with local resources
+func TestScriptBasedDownloadAndInstallLocalResources(t *testing.T) {
+	tempDir := "tempdata"
+	assertDirs(t, tempDir, true)
+	defer os.RemoveAll(tempDir)
+
+	installScriptAlias, installScriptBody := storage.GetTestInstallScript()
+	installScriptPath, installScriptHash := createLocalArtifact(t, tempDir, installScriptAlias, installScriptBody)
+	localResourceAlias, localResourceBody := "local.txt", "test"
+	localResourcePath, localResourceHash := createLocalArtifact(t, tempDir, localResourceAlias, localResourceBody)
+
+	t.Run("Test_without_copying_artifacts", func(t *testing.T) {
+		artifacts := []*hawkbit.SoftwareArtifactAction{
+			convertLocalArtifact(installScriptPath, installScriptAlias, installScriptHash, len(installScriptBody)),
+			convertLocalArtifact(getAbsolutePath(t, localResourcePath), localResourceAlias, localResourceHash, len(localResourceBody)),
+		}
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{}, artifacts, modeLax, "", true)
+		checkFileExistsWithContent(t, filepath.Join(tempDir, "status"), "message=My final message!") // install file was executed in correct folder
+	})
+
+	t.Run("Test_with_correct_install_path", func(t *testing.T) {
+		artifacts := []*hawkbit.SoftwareArtifactAction{
+			convertLocalArtifact(installScriptAlias, installScriptAlias, installScriptHash, len(installScriptBody)),
+			convertLocalArtifact(getAbsolutePath(t, localResourcePath), localResourceAlias, localResourceHash, len(localResourceBody)),
+		}
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{tempDir}, artifacts, modeLax, "*", true)
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{tempDir}, artifacts, modeStrict, "*", true)
+	})
+
+	t.Run("Test_with_no_install_path", func(t *testing.T) {
+		artifacts := []*hawkbit.SoftwareArtifactAction{
+			convertLocalArtifact(installScriptPath, installScriptAlias, installScriptHash, len(installScriptBody)),
+			convertLocalArtifact(getAbsolutePath(t, localResourcePath), localResourceAlias, localResourceHash, len(localResourceBody)),
+		}
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{}, artifacts, modeLax, "*", true)
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{}, artifacts, modeStrict, "*", false)
+		testScriptBasedSoftwareUpdatableOperationsLocal(t, []string{}, artifacts, modeScoped, "*", false)
+	})
+
+	t.Run("Test_with_other_install_path", func(t *testing.T) {
```

Comment:
- installPath was renamed to installDirs, the name has to be adapted
- "other install path" is not very informative, more concrete name can be adapted